docs/guidelines/local-development.rst

```diff
@@ -42,27 +48,68 @@ repository locally by executing:
 
   $ git clone https://github.com/cekit/example-common-module.git ~/repos/example-common-module
 
-Then we will create ``override.yaml`` file next to the ``image.yaml``. Override.yaml should look like this:
+Then we will create ``override.yaml`` file next to the ``image.yaml`` with following content:
 
 .. code-block:: yaml
 
   modules:
     repositories:
-      - path: "/home/user/repo/cct_module"
+      - name: common
+        path: /home/user/repo/cct_module
 
-Now we can build the image using overridden module by executing:
+Now we can build the image with Docker using overridden module by executing:
 
 .. code-block:: bash
 
-  $ cekit generate --overrides-file overrides.yaml
+  $ cekit build --overrides-file overrides.yaml docker
+
+.. note::
+    Instead using an overrides you can use inline overrides too!
+
+    .. code-block:: bash
+
+        $ cekit build --overrides '{"modules": {"repositories": [{"name": "common", "path": "/home/user/repo/cct_module"}]}}' docker
+
+When your work on the module is finished, commit and push your changes to a module repository so that
+other can benefit from it. Afterwards you can remove your overrides file and use the upstream version of the module
+again.
+
+Notes
+^^^^^^
+
+Below you can see suggestions that should make developing modules easier.
+
+Always define name for module repositories
+*******************************************
+
+We use the ``name`` key as the resource identifier in all places. If you do not define the ``name``
+key yourself, we will generate one for you. This may be handy, but in cases where you plan to use
+overrides it may be much better idea to define them.
```

Comment:
Question - is the generated name unique?

Reply:
It is not really a unique name. It's based on the resource provider data. In case of git repositories it is the name of the repo IIRC, bit I would need to double-check in the code.

Reply:
I was right: https://github.com/cekit/cekit/blob/f75f098a74be1b50ec58fcbb195a49a0222ddcde/cekit/descriptor/resource.py#L288-L290

So, for `https://github.com/cekit/cekit.git` it will result in `cekit.git` as the `name` key.